### PR TITLE
feat: handle streaming control rpc in worker loop

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Previously we have a `LocalStreamManagerCore` that keeps track of actor states. The `core` is wrapped by `Arc<Mutex<...>>` and shared by grpc handlers. To handle each grpc request, we will have to acquire the lock on `core` to fetch or modify the states.

In this PR, the `LocalStreamManagerCore` is renamed to `StreamActorManagerState` and owned by `LocalBarrierWorker` loop so that we don't need to acquire lock to access it. All streaming control rpc request will be sent to `LocalBarrierWorker` as an event to handle the request. `LocalStreamManager` is renamed to `StreamActorManager` to handle actor creation, and the original `LocalStreamManager` becomes a simple wrapper on local barrier event sender to send grpc request to barrier worker as event.

Previously handling all streaming control rpc except `build_actors` are non-async. Therefore in `LocalBarrierWorker` loop, handling these events can easily be non-blocking. For `build_actors` rpc, since it is async when creating the actors, to handle it in `LocalBarrierWorker` loop, it becomes a two-phase operation. Its async part that creates the actors will be spawned to run concurrently so that it won't block the worker loop. The worker loop will keep polling the join handle to get notified on the completion of creating actors. After actors are created, it will spawn the actors and send the response to the original rpc handler.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/stream/src/task/stream_manager.rs [hunk 1]
```diff
@@ -251,26 +216,62 @@ impl LocalStreamManager {
...
+    pub async fn update_actors(&self, actors: Vec<stream_plan::StreamActor>) -> StreamResult<()> {
+        self.local_barrier_manager
+            .send_and_await(|result_sender| LocalBarrierEvent::UpdateActors {
+                actors,
+                result_sender,
+            })
+            .await?
+    }
+
+    /// This function could only be called once during the lifecycle of `LocalStreamManager` for
```
Comment (kwannoel) [inline src/stream/src/task/stream_manager.rs:250]: Is there some way we can enforce that this function is only called once at most?

File: src/utils/futures_util/src/misc.rs [hunk 2]
```diff
@@ -76,3 +77,39 @@ pub async fn await_future_with_monitor_error_stream<T, E, F: Future>(
         Either::Right((output, _)) => Ok(output),
     }
 }
+
+pub struct AttachedFuture<F, T> {
```
Comment (kwannoel) [inline src/utils/futures_util/src/misc.rs:81]: Don't quite get this, can we add some docstring on its purpose?

File: src/stream/src/task/stream_manager.rs (hunk 1 above)
Comment (wenym1) [reply src/stream/src/task/stream_manager.rs:250]: I think the doc string is outdated. I will removed it.

File: src/utils/futures_util/src/misc.rs (hunk 2 above)
Comment (wenym1) [reply src/utils/futures_util/src/misc.rs:87]: I have adde the docstring. 

In short, `AttachedFuture::new(future, item)` is similar to `future.map(|output| (output, item))`. The only difference is that the `AttachedFuture` provides method `into_inner` to return the inner future. In practice, we will wrap the join handle with `AttachedFuture` as a future to pass to `FuturesUnordered`. In recovery, we will drain all futures and take the inner join handle to call abort on it to abort and join the future.

**Review by yezizp2012 — APPROVED** [review]
LGTM

File: src/stream/src/task/stream_manager.rs [hunk 3]
```diff
@@ -251,26 +216,62 @@ impl LocalStreamManager {
...
+    /// now.
```
Comment (kwannoel) [inline src/stream/src/task/stream_manager.rs:251]:
```suggestion
```

**Pull request merged**